x/ecocredit/basket/msg_create_test.go

```diff
@@ -103,15 +102,9 @@ func TestMsgCreateValidateDateCriteria(t *testing.T) {
 		d   DateCriteria
 		err string
 	}{
-		{"nil-min_start_date",
-			DateCriteria{MinStartDate: nil},
-			"unsupported date_criteria value"},
 		{"bad-min_start_date",
 			DateCriteria{MinStartDate: &gogotypes.Timestamp{Seconds: time.Date(1400, 1, 1, 0, 0, 0, 0, time.UTC).Unix()}},
 			"date_criteria.min_start_date must be after"},
-		{"nil-start_date_window",
-			DateCriteria{StartDateWindow: nil},
-			"unsupported date_criteria value"},
```

Comment:
"nil-min_start_date", "nil-start_date_window" will fail with the changes in this PR, since `DateCriteria{MinStartDate: nil}.String() == ""`  will be true.

Reply:
In fact empty DateCriteria is invalid. We should use `nil` instead for the whole parameter.
I will update this PR:
* rollback the test and validation changes
* Update CLI to use nil as DateCriteria, if it was not defined.